File: android/app/src/main/java/com/englishfriendai/app/domain/model/Message.kt
package com.englishfriendai.app.domain.model

enum class Sender { USER, AI }

/** Language mode the user has selected for the current conversation turn. */
enum class ConversationMode { ENGLISH, TAMIL_ENGLISH, TAMIL }

data class Message(
    val id: String,
    val conversationId: String,
    val sender: Sender,
    val englishText: String,
    val tamilTranslation: String?,
    val timestamp: Long,
    val correction: Correction?,
    val audioUrl: String? = null
)

File: backend/src/models/Message.js
'use strict';

const mongoose = require('mongoose');
const { Schema } = mongoose;

const messageSchema = new Schema(
  {
    conversationId: {
      type: Schema.Types.ObjectId,
      ref: 'Conversation',
      required: true,
      index: true,
    },
    sender: { type: String, enum: ['user', 'ai'], required: true },
    englishText: { type: String, required: true },
    tamilTranslation: { type: String, default: '' },
    audioUrl: { type: String, default: null },
    createdAt: { type: Date, default: Date.now },
  },
  { timestamps: false }
);

messageSchema.index({ conversationId: 1, createdAt: 1 });

module.exports = mongoose.model('Message', messageSchema);
